### PR TITLE
[Cult 4] "NAR-SIE'S HOLIDAYS" , aka "The End of Cult 3.0"

### LINKED ISSUE #20705: Cult 3 tome background doesn't load sometimes
image

#### Steps to reproduce
- Be cultist
- Open book
- Browse the pages
- Pic related

### LINKED ISSUE #20706: Cult 3 tome is hard to read
#### Description of issue
image
It may be due to my shitty monitor but there's something about brown text on a brown background that makes it tough to read in certain parts.

### LINKED ISSUE #23031: Cult Act 3 needs a rework - Here's my suggestion
Screenshot 2019-05-25 00 18 49

Tell me. What do you see in this picture?

Suffering? Misery? The dying wishes of a hundred souls, torn in anguish yet heard by none?

I tell you what I see in this miserable depiction. I see a rework.

### THE REWORK FOR ACT 3 - THE BLOODBATH

Blood tiles are awful. Instead, let's replace it with a summoning ritual.
After doing the sac, the bloodstones shall spawn as normal. One of these bloodstones will serve as the anchor, the place where Nar'sie will rise. It has more HP than the others and should also glow or something. Nar'sie will be summoned in 10 minutes if this stone is not destroyed. It will also not be the original stone, to prevent space bases.
The 4 other bloodstones serve as tools to fasten the process, reducing the time by 1.5 minutes each, down to 4 with all of them active. Once it's fully charged, act 4 proceeds as normal.

Thoughts?

### LINKED ISSUE #23048: Bloodstones are a tad extra bullshit on roid
(WEB REPORT BY: chazach REMOTE: 206.221.180.138:7777)
# Revision
bc187937068fa7fb30a51cf9cb9ba0c1614ef4bc

# Description
as well as spawning on the edge of nowhere they also delete some roid tiles and turn them into space

### LINKED ISSUE #24226: Cult really shouldn't be able to fire on snowmap
#### Description of issue

Snowmap has a TONS of places to hide, perhaps even more than roid and deff cult. The impossibility of raising an altar in the roid is null and void if you can just raise one in bumfuck nowhere in the woods

Relying on sufficient maint coverage is made impossible unless you have 5+ officers, given how far away everything is ; same for responding to calls or raiding departments

I don't even want to begin to imagine what would be bloodstone hunting on snowmap

### LINKED ISSUE #26811: I don't like cult

### LINKED ISSUE #29558: Cultist Tome Unreadable
(WEB REPORT BY: punishedpajeet REMOTE: 206.221.180.138:7777)
# Revision
88ffb70118d9dfc7d757e0d0ead628f9a8955a45

# Description
Every other time I open the cultist tome, it has a background that makes it totally unreadable. 1st time is unreadable, 2nd is blank background (readable), 3rd is unreadable, etc. The text also doesn't wrap according to the window

https://imgur.com/a/hJ0tZ3l

# Steps to Reproduce
Make a book using the See Blood Hell runes and open it

# What you Expected
Legible text with proper word wrapping

# What Actually Happened
Scribbles and chickenscratch

### LINKED ISSUE #29836: Shrunken vampire head doesnt detect cult runes.
The shrunken head glows when it detects blood nearby, however it doesn't glow when cult runes are nearby even though they are made out of blood.

### LINKED ISSUE #30383: Security HATES him, try this one weird trick to succeed in every single fight as a cultist!
With the pale body and stun talisman you can use the talisman in-hand and stun everybody surrounding you for 5 seconds. This cannot be blocked and is nearly impossible to prevent because it requires being nearly offscreen from the target to not be affected (especially with our run speed). This has been the case for the entirety of cult 3 as far as I know.

### PR DESCRIPTION
**Pull request by DeityLink**
Fixes #20705
Fixes #20706
Fixes #23031
Fixes #23048
Fixes #24226
Fixes #26811
Fixes #29558
Fixes #29836
Fixes #30383

I've done a few Cult 4 PRs earlier this year but this one is actually tackling the *meat* of cult as an antagonist faction.
I started working on this PR properly almost precisely a month ago, although you could argue that all the work I did on mind UI before that was in preparation for this moment as well. So anyway here we go.

**UPDATE: This PR is finally ready for review 24 days later. Cult 4 took a total 21 days of coding, and probably as much discussing cult over the years with players, and coming up with this design. I don't know if this will be the last iteration of cult, but I think it will leave the antagonist in a much better place than the one perpetuated by Cult 3.0**

**I'll focus on fixing the inevitable new cult bugs that will crop up and addressing potential design issues for a while before I decide to get started on the next big parts of Cult 4.**  

-----

<h1 align=center>Cult 4: Upheaval Update</h1>

<h3>1. ACT system removal </h3>

The ACT system one Cult 3.0 was an attempt at furthering the new required objectives from Cult 2.0 into some narrative that also helped new players discover runes progressively. This brought a ton of problems when new cultists were expected to meet up and raise the Altar without having any of their other cult tools available, and caused the runes unlocked in later Acts to be completely forgotten about. On top of that, the cult's ultimate objective being to raise Nar-Sie no matter what happens prevents the crew from ever being able to take it easy until they know for sure the cult has been exterminated, this is what forces Security to deal with cult with extreme prejudice, and even call the shuttle pre-emptively. And that's not counting the fact that having the exact same Act-based "narrative" for the past 3 years has led to most cult rounds feeling very samey.

So for the time being, until the Ritual system is implemented, which will be non-linear and allow cultists to choose their own objectives, **Cult is being reworked from an objective-based antagonist to a completely freeform antagonist**. I'll be paying close attention to how cults operate, and how the crew handles them.

As a consequence of the Act System's removal, the following changes are de facto in place:

* **The Cult has no Raise Altar, Convert or Sacrifice objectives anymore**
* **The Bloodstone phases are gone, so the Cult can no longer raise Nar-Sie (hence, "Nar-Sie's holdays")**
* **Cultists reject implants and react to holy water the same way at all stages of the round (same as they did before on Act 2)**
* **Cultists have access to all runes from the beginning**
* **Ghosts can no longer write in blood while invisible since there is no Act 3 (they still can while made visible by Arcane Tomes)**

<h3>2. Tattoos removal</h3>

Since there are no more Acts, **there are no more tattoos to unlock**. Some of them (Blood Dagger and Shortcuts for sure) are planned to come back as rewards for the Rituals that will come in the next update.

<h3>3. Cult Comms</h3>

spire

Since there is no more Cult Chat tattoo, **Cultists will be able to use :x to communicate as long as there is a Spire on the Z-Level**. Spires have been repurposed from Tattoo dispensers to Cult Comms Relays. In practice that means cult-wide comms should become more readily available since the Spire can also be built right away.

<h3>4. Cultist Roles</h3>

The cult thrives when its members cooperate, and although we've experimented with the idea of a Cultist Chief, it being randomly assigned means that over half the time the designated chief is a newbie, or someone who cannot lead the cult for other reasons. I've long believed that leaders should arise organically among the cult, and for now a new system has been introduced:

* **When they spawn or get converted, Cultists can pick a role that serves to signal to the rest of the cult how they want to play**
* **"Herald" is the role closest to what regular cultists were.** They can do whatever they think will help the cult best (even if that means just doing their job as crew member). Ideally they would keep the crew and security busy enough to take heat off Acolytes and Mentors.
* **"Mentors" are expected to train Acolytes, and cooperate with other mentors to give the cult some purpose**. When the cult does well, it's in general because some cultist takes the lead to give other cultists some strategy, that's basically what is expected of this role. There's no more objectives in Cult right now, so they're also gonna require some creativity.
* **"Acolytes" are the newbie role, for people either new to cult or who just want to toy and experiment with runes and stuff.** The game will automatically tries to assign them to a mentor, informing both that they are now part of a mentorship.
* **Cultists can switch role every 5 minutes. Doing so ends any ongoing mentorship**

<h3>5. Newbie tips</h3>

image

*Because it is well known that most new cultists generally don't go on the wiki*, **Acolytes have an extra button on their UI that lets them read some laconic basic tidbits on what cultists can do**, so they have some idea about their role even if no mentor is available to take care of them. Obviously it's by no means a tutorial, but it should do its job at giving players motivation to want to learn more about it, and it is by no means a replacement for Mentors to do their job, but most new players should appreciate not being completely on their own.

You can check the list of tips here: https://imgur.com/a/3EpGTiq

<h3>7. Cultist Cap</h3>

* **Cultists can no longer have more than 9 human players**. Further conversion attempts automatically cuff the target instead, you get a warning message during the conversion.
* **The first construct of each type do not count toward the cap, but subsequent ones do.** So if you create 2 juggernauts you reduce the human cap to 8, but if there's no artificer you'll always be able to create one.
* **Shades do not count toward that cap, but dead human cultists do.** If a human cultist disconnects without warning for an extended period of time, feel free to shade their body. Don't worry, they can get it back later (see Rune tweaks).
* **No Cult-related announcement have a chance to play anymore after a conversion occurs.**

<h3>8. New Cult UI and Rune-Writing</h3>

image

* **Cultists have a new UI on their screen that lets them write and erase rune.** This UI can be moved anywhere on the screen by holding its button on the right.
* **Cultists now have access to the full rune list from the start**, and Arcane Tomes no longer change anything when writing runes. There are two buttons you can choose when you want to write a rune. One opens a dial that lets you write words one by one without delay, the other opens the list of all runes (like the old Write Runes spell would when holding an arcane tome) and selecting one opens the dial and highlights the words you have to press, both allowing you to write runes faster than before without having to hold a tome, AND to memorize patterns so eventually you may not even have to waste time opening the list at all.
* **Runes have new enhanced sprites and animations when created or erased**. Runes now look much more like Blood splatters, until you activate one and the black marks appear on them.
* **Runes that haven't been activated yet (and don't have the black marks) can be cleaned off like any other blood splatter can (soap, mop, space cleaner)**. Activated runes still require salt, or holy salts.

image

<h3>9. Soul Blade Tweaks</h3>

* **Soul Blades are now "connected" to the last cultist that held them, and they can move without spending blood while they are within 4 tiles of them. In that radius, they also benefit from the same blood gauge regeneration they would get if they were being held.**
* **Soul Blades have had their attack speed fixed so it's always exactly 1 hit per second, instead of sometimes more, sometimes less.**
* **For the first 0.4s of their attack, Soul Blades can now reflect projectiles**

As an unrelated bugfix, Rockets no longer explode if they are reflected 😌

<h3>10. Altar Tweaks</h3>

altar

**The "Commune with Nar-Sie" button on the altar now lets you send messages to Nar-Sie (to admins) just like Comms console lets you contact Centcomm or the Syndicate.** Complete with a button to punish those who abuse it (or who want an amazing death).

<h3>11. Rune Tweaks</h3>

image

* Raise structure now has the same blood cost whether the cultists using the rune are wearing robes or not. Robes are now mostly intended for both protection and anonymity. (They still speed up conversion)
* **Summon Tome rune and Conjure Talisman rune have been merged into a single rune called Paraphernalia**. This rune lets you produce all sorts of cult apparatus for a small blood cost (See pic above). This rune is based on the Conjure Talisman rune, and can still be used to fax talismans to other cultists' tomes.
* Among its items, Paraphernalia can let you spawn a new snazzy looking candle that lasts for a whooping 20 minutes when lit up. You can now also lit up candles by using them on one another no matter which is lit. Those candles will have more purposes in the Rituals Update coming next.
* The old Summon Tome rune is now hidden. It can still be written and activated to produce a tome if you remember the words, but won't appear on the full rune list or the arcane tome. Same with the Stream rune (which can also still be activated if you know the words, mostly to prevent newbies from accidentally out themselves while testing runes.)
* The AoE version of the Stun runes and talismans now hits mobs around it in a circle rather than a Square. It now takes 1 extra second to go off after being cast, an a mark appears on the ground covering the whole area that will be hit by it, giving players a much more fair chance to dodge it.
* Cult Hoods now count as helmets in regard to Flashbangs (to be fair, they cover ears unlike many helmets). Dress up appropriately when confronting security.
* Blood Magnetism now fails to either summon or rejoin a cultist that has holy water in their body, or a holy implant.
* **Resurrect rune has been reworked, now called Reincarnation. It lets shades become humans again.** Or to be more specific, it gives the mob whatever body it originally had when its mind was first initialized. Be careful if you were a plasmaman, as you'll be dressed in nothing but rags. The rune no longer requires ingredients other than its hefty blood cost, and no longer works on ghosts at all. Humans reincarnated have a very particularly clear skin and hair colour that cannot be obtained normally, and might need a UI injector to appear less conspicuous. (Reincarnated humans also have some cult tattoos on their skin, those don't provide any other gameplay changes)

<h3>12. Other changes</h3>

* Shrunken Vampire Heads (the Trader item) now detects runes as well.

-----

This is technically Part 6 of Cult 4, and Part 1 of the Upheaval update which drastically changes the flow of cult rounds.
Part 2 will introduce the Ritual system I mentioned in the Cult 4.0 doc (see #26941), **when I get to it someday.**

I hope you will enjoy this new iteration of cult, as always feel free to share your experience and report bugs so we can improve on it.

:cl:
* tweak: Some minor cult changes. Check PR 30672 on Github for a more extensive changelog.

### CODE REVIEW

**Comment by cadaverouscoyote** [comment]
just wanted to say that indie game tutorial thing you showed off is unnecessary and i honestly abhor it. between the wiki page, their converter, and a tome, a new cultist should have all the info they need to get started; if they don't, that's what needs fixing. it's also not going to promote team play if the converter runs off to handle other business while expecting the noob to sit in place and read through the tutorial on their own. another, massively understated issue is that nobody learns without motivation, and tossing an obtrusive info-dump at a prospective new cultist isn't going to make them actually want to engage with what appears to be an excessively bloated team mode so unintuitive as to require an in-game tutorial overlay, whose mechanics are out of line with what they already know, where one fuck up on their part can screw over the rest of their team. it embodies every existing problem and doesn't solve any of them

also why would you make it dominate the game view when everything could be off in a window what the fuck

**Comment by DeityLink** [comment]
> just wanted to say that indie game tutorial thing you showed off is unnecessary and i honestly abhor it. between the wiki page, their converter, and a tome, a new cultist should have all the info they need to get started;

That's the problem. Roundstart cultists are often 1st time cultists as well and have no converter (or even experienced team mate at times) to teach them the ropes. I don't want cultists to have to stop by the wiki page, going on the wiki should be something you do to deepen your understanding of the antag, not to get started, which is what the ""tutorial"" is about. 

>If they don't, that's what needs fixing.

Such as by adding an optional starting tips pop up? To be clear, it's a last resort for newbie cultists who don't have anyone to teach them which honestly shouldn't happen too much as some players already said they wanted to roll as mentors. (Which is the main fix to that problem, the help pop up is just a bonus that you can ignore altogether)

>It's also not going to promote team play if the converter runs off to handle other business while expecting the noob to sit in place and read through the tutorial on their own.

Just so we're clear people have been doing that since forever and they won't stop it unless I point a gun at them lol. More seriously once again those are starting tips, not a full tutorial, and I don't think anyone expects to let newbie cultists handle themselves with that alone as a guide.

> another, massively understated issue is that nobody learns without motivation

That's true, the motivation here is the ability of using new powers in a different way than wizards or others do, which is what the first tip is about.

> and tossing an obtrusive info-dump at a prospective new cultist isn't going to make them actually want to engage with what appears to be an excessively bloated team mode so unintuitive as to require an in-game tutorial overlay, whose mechanics are out of line with what they already know

Well new cultists aren't expected to be productive right out the gate. Without cult objectives they can learn at their leisure, and if learning an antag this intricate isn't your cup of tea then maybe cult just isn't for you.

> where one fuck up on their part can screw over the rest of their team. it embodies every existing problem and doesn't solve any of them

Well you probably don't know because I can't write the full changelog until tomorrow but cult is completely changed design wise, there's currently no more objectives, nar-sie to raise, and the cult is expected to make itself known as the purpose is now more to mess with the crew, so a noob cultist outing the cult isn't much of a problem anymore.

> also why would you make it dominate the game view when everything could be off in a window what the fuck

Just so you know you can click anywhere on that window to close it. 

Please wait tomorrow for the full changelog instead of focusing on the one small feature that someone posted in the thread.

**Comment by DeityLink** [comment]
And I do hope you'll give me your thoughts again then. I understand your frustration, and feedback even from people who hate the way I do things gives me valuable insight, especially when it's properly constructed, and it's easier to have a conversation here than in the thread.

**Comment by BomberBro** [comment]
>just wanted to say that indie game tutorial thing you showed off is unnecessary and i honestly abhor it. between the wiki page, their converter, and a tome, a new cultist should have all the info they need to get started; if they don't, that's what needs fixing. 

While I agree in principle the sad fact of life is that this game has absolutely abhorrent tutorial mechanics (AKA zero), altclick is an invaluable tool for a lot of machines and mechanics and gets mentioned exactly nowhere, most items don't have any text in their descriptions indicating that you can alt click or right click them for additional actions, etc. Even the basic controls aren't really displayed anywhere obvious. If this helps cult become less hellish and rounds in general more fun then I don't care if it's straight out of CalArts. 

I'd rather we have something like this for a tutorial though, like 1-5 connection accounts getting a little button that gives them 10ish slides of controls and interactions while explaining the HUD, it'd probably make the whole onboarding process for a 20 year old game better.

**Comment by DeityLink** [comment]
> I'd rather we have something like this for a tutorial though, like 1-5 connection accounts getting a little button that gives them 10ish slides of controls and interactions while explaining the HUD, it'd probably make the whole onboarding process for a 20 year old game better.

@BomberBro so now I know what I'm doing Sunday. *Mind UI intensifies*

**Comment by cadaverouscoyote** [comment]
i did skim over the google doc and since anything may have changed, i decided for now to only give my thoughts on what i've seen myself, that being the one small feature that i posted in the thread

> Roundstart cultists are often 1st time cultists as well

the players without any experience with cult being the ones to opt into playing cult is an entire other problem itself, which is the symptom of yet another problem, well beyond the scope of a qrd on the mode's mechanics. in any case, the fact is that first time starting cultists are naturally going to click the wiki button next thing, because they have a lot to do and they're going to want more specific info. the whole thing can be streamlined by just having a decent introduction in the wiki page

> Such as by adding an optional starting tips pop up?

that isn't what i said, though it may have been hard to parse. "that" is specifically the deficiency between everything i mentioned, which is what would need to be fixed in the first place

> To be clear, it's a last resort for newbie cultists who don't have anyone to teach them

a big animated arrow that says "click" over a fat button doesn't give me the impression of a last resort. a new player's not going to ignore it, it's gonna be the first thing that grabs their eye. the converter will have to try to pull their attention away from the tutorial, which may leave the newbie feeling overwhelmed

> Just so we're clear people have been doing that since forever and they won't stop it unless I point a gun at them

yeah. like i said, every existing problem

> if learning an antag this intricate isn't your cup of tea then maybe cult just isn't for you.

cult not being people's cup of tea was the root of the boxing issue

> Just so you know you can click anywhere on that window to close it.

why is that necessary? why not just have it be in a separate window so you can move it anywhere that isn't obstructive so you can glance between it and whatever's happening in game instead of having to close and open it again depending on where you wanna be able to see? i get that mind ui's the hot new shit and the possibilities are exciting and all but please have restraint

i'm still not convinced the tutorial serves an effective purpose. you raised some points based on some big promises which are based on a bunch of different ideas that you find to be good in theory, and the only response i can give to that thinking is this: i'm sorry but i can't hold any confidence that this isn't going to be cult 3.0 2.0, and i don't think these excessively ambitious overhauls are good practice in general

**Comment by DeityLink** [comment]
> that isn't what i said, 

I wasn't quoting you, I was indicating that this was the answer I choose to alleviate a problem which will exist no matter what else I do. (An answer you don't like I get that)

> a big animated arrow that says "click" over a fat button doesn't give me the impression of a last resort. a new player's not going to ignore it,

You make a good point, I'll make the arrow only appear if no mentor was found.

> in any case, the fact is that first time starting cultists are naturally going to click the wiki button next thing, because they have a lot to do and they're going to want more specific info.

You think that because you personally know how complex cult is, a new player might not, and I don't think *everyone* naturally opens the wiki when they get a new antag. Most are fairly straightforward (and to clarify, I don't think cult *should* be straightforward. Cultists teaching each others is one of the things I like most about cult)

Also if you're the one who posted the webm and you looked at the full tips you'll notice how laconic and superficial it is. It's by no way a tutorial but it does its job of informing players of how complex cult actually is so they can *then* start actually diving into it. In short I'd rather cultists go to the wiki thinking "gosh I want to know what runes there are" or "how do I make constructs" than "gee, what do you even *do* as a cultist anyway?" Which is a situation I've encountered time and time again. So many ahelps of players going "Help, how do I cult?"

Adding a wiki link at the end of the tooltips might also lead into that now that I think about it.

> cult not being people's cup of tea was the root of the boxing issue

Before the boxing issue there was the "convertee ignores the cult and goes back to jobbying instead of helping the cult" issue, which is also why by letting cultists target players based on their preferences with the Seer rune update, and capping the cultist count to 9, has become much less prevalent.

> why is that necessary?

Same reason I prefer in-game holomaps to station renders in browsers, I like integrated information more and pop-ups less. Makes it feel a bit more like the game wasn't made in the early 00s. Also I'm fairly sure most actually *new* players like those better but I don't have stats to back that up, just discord feedback (where people usually don't hesitate to tell me when my ideas suck ass)

> you raised some points based on some big promises which are based on a bunch of different ideas that you find to be good in theory, and the only response i can give to that thinking is this: i'm sorry but i can't hold any confidence that this isn't going to be cult 3.0 2.0, 

Game design is an iterative process and I am continuously learning. I can only use experience and feedback to try and do it better each time. That's how it actually works in the video game industry out there, except unlike those I don't have the luxury of affording teams of QA Testers and playtesters to gather feedback (that's also why there are so few successful MMOs, those are nigh impossible to test properly beforehand). I certainly do not promise that this version of cult will magically solve *every* problem, but I'll keep adjusting and try again because that's what game dev is about, and each time it's a bit better. 

> and i don't think these excessively ambitious overhauls are good practice in general

Good practice for what? So far big overhauls have proven great at bringing back players. Heck Virology attracted a lot of new Virologist players after its, and even xenoarch is more popular than xenobio right now (xenobio badly needs new content). Give me some examples of good practice so I can put your take in perspective.

**Comment by cadaverouscoyote** [comment]
> I wasn't quoting you, I was indicating that this was the answer I choose to alleviate a problem which will exist no matter what else I do. (An answer you don't like I get that)

you didn't quote me, you misunderstood what i tried to say and went into something beside my point, which i'm about to reiterate so nvm

> You think that because you personally know how complex cult is, a new player might not, and I don't think _everyone_ naturally opens the wiki when they get a new antag.

the wiki button's on screen at all times, a link to the wiki is in the top-center of the changelog window that opens when you first join, the server welcome message in the text box commands you to "Read the wiki!" in like size 30 text with yet another link, and the standard for every job is to peep the wiki before you begin. no new player isn't aware of the wiki and everyone's encouraged to use it, so if they have a question that they can't pose to somebody ic or to an admin, it's naturally going to be their next choice

> Also if you're the one who posted the webm and you looked at the full tips you'll notice how laconic and superficial it is. It's by no way a tutorial

yeah i'm just calling it that to annoy you

> In short I'd rather cultists go to the wiki thinking "gosh I want to know what runes there are" or "how do I make constructs" than "gee, what do you even _do_ as a cultist anyway?" 

the deal is that since they're going to go to the wiki anyway, what to even do as a cultist should be covered by the first paragraph. there's no point to not presenting everything as concisely as possible and cutting out distracting, superfluous additions

> Same reason I prefer in-game holomaps to station renders in browsers, I like integrated information more and pop-ups less. Makes it feel a bit more like the game wasn't made in the early 00s.

problem is that it's straight up less convenient for the player to handle while offering no real benefit. it's like every modern appliance needing buttons that are completely smooth and flush with the surface of the thing, with no tactile feedback whatsoever, just to imitate the limitations of a touch screen because that's the only thing manufacturers think people are comfortable with these days. it doesn't even have an ic reason for it to be that way, all the info is presented ooc, nobody's immersion is being enhanced by doing it like this. you think a window's too old-fashioned, but if this isn't better in every measure it isn't a solution, it's just change for the sake of change. ss13 does not need to be like every other game. i get that you're excited to work with mind ui and you like how this turned out, but that doesn't make its inclusion necessary or a good idea

> Game design is an iterative process and I am continuously learning. I can only use experience and feedback to try and do it better each time.

"one step back, two steps forward" is the right idea. the problem with ground-up overhauls is potentially ending up a couple hundred steps back from where you started. our players run the gamut of playstyles and knowledge levels in a game so mechanically convoluted that when it comes to the meta the only way to know for certain how anything will work in practice is to throw it into their enclosure and see what happens. experimentation is the name of the game, and experimentation entails controls. i know it's the fun way to do things, but laying big plans is setting yourself and the playerbase up for disappointment

> Good practice for what? So far big overhauls have proven great at bringing back players.

i would not take population fluctuation as strong indication of anything; i think player retention is the better metric. i didn't say that everything-at-once overhauls will always fail, but they certainly carry the highest, most unnecessary risk of failing catastrophically

> Give me some examples of good practice so I can put your take in perspective.

atomicity, test merges followed by server polls, use of the test server, a willingness to discard ideas, and wariness of sunk-cost fallacy. high ambitions are always gonna grind against those last two to some degree imo

this conversation got away from the topic i wanted to cover and into stuff i didn't want to touch on before the changelog dropped, but that's kind of the nature of the big promises thing

**Comment by DeityLink** [comment]
> the wiki button's on screen at all times, a link to the wiki is in the top-center of the changelog window that opens when you first join

So is the bug report button yet only a few people ever click it. Most players never check that area of the screen. People even ahelp about the map's name at times. And once again that's beside the point: which is that most players don't go on the wiki blind, and I argue they shouldn't.

> yeah i'm just calling it that to annoy you

So you're purposefully misrepresentating your own arguments for the sake of annoying people instead of trying to make them understand your point? Well forgive me for misunderstanding your points then lol.

> the deal is that since they're going to go to the wiki anyway, 

They don't, that's my entire point and i've seen countless ahelps over years by clueless players to confirm it. (And that's not counting the thread schizos who say they're purposefully not checking the wiki)

> problem is that it's straight up less convenient for the player to handle while offering no real benefit

Well ok let's list a few then.

* Those interfaces don't lag like regular windows popup usually do
* Unlike NanoUI you won't have to restart your game because the popup goes blank for no reason
* You can easily have them display any sprite from the game, including animated ones, something other UIs can't do.

But just call it "change for the sake of change" when it's not even changing anything, it's an addition, one that you're free to disregard altogether and that an old player like you won't even have to see at all since I don't expect you to pick the beginner role.

> i know it's the fun way to do things, but laying big plans is setting yourself and the playerbase up for disappointment

It's not "fun" that's why it took me so long to finally make this PR. It's the result of years of getting years of player feedback and pitching ideas at them. Many of which even came from the thread.

> the problem with ground-up overhauls is potentially ending up a couple hundred steps back from where you started.

So because a new overhaul is potentially bad we should never try to fix bad features? No, we keep making changes until we reach a desirable result. Better than dropping a big feature and fucking off without ever adjusting it. That's what some of the complaints about Cult 3.0 and that's why this PR is reverting the ACT system among other things. When people hated black market and its coder fucked off we reverted it. Most people hate the ACT system, please tell me why it's bad to change it.

> experimentation is the name of the game, and experimentation entails controls.

How is that not what I'm doing? Or do you have a better idea of how to do it and plan on doing it yourself? If I don't do it who else will do it? Shifty believed in the potential of cult 2.0 and attempted to preserve it as Legacy Cult which has never properly worked and has remained unmaintained for years.

> i didn't say that everything-at-once overhauls will always fail, but they certainly carry the highest, most unnecessary risk of failing catastrophically

I'll argue that the worst potential catastrophic fallout would still be limited to cult, and easy to contain even by just disabling the ruleset. You're making this sound apocalyptic when it really isn't.

>  use of the test server, 

Shifty struggles getting people to come there just to test Europa Lights. Just like for the MMOs I talked about earlier we won't get usable feedback unless the changes are tested in actual context on the main server. Space Parallax, another big overhaul I worked on worked fine on my machine, but terribly when 30 players moved around, and we just did the necessary optimizations until it worked, but just like Europa Lights the real valuable tests only happen on the main server

> willingness to discard ideas, and wariness of sunk-cost fallacy. 

Considering most of this PR consisted of deleting code I wrote 3 years ago I think I can get over that just fine.

Thank you for expressing your concerns, I'm coming home soon and the changes will be listed properly within the next few hours

**Comment by DeityLink** [comment]
Aaaand it's finally up.

**Comment by KingofHONK** [comment]
> just wanted to say that indie game tutorial thing you showed off is unnecessary and i honestly abhor it. between the wiki page, their converter.....

Uhhh...have you played this game? How do you learn engineering, or botany, or anything else? If you want to learn anything about this game at all you have to read a wiki page or learn from someone else. Saying "thing in ss13 shouldnt be so complex in the first place" is like saying a hot dog is a sandwich.

**Comment by DeityLink** [comment]
> 
> 
> > just wanted to say that indie game tutorial thing you showed off is unnecessary and i honestly abhor it. between the wiki page, their converter.....
> 
> Uhhh...have you played this game? How do you learn engineering, or botany, or anything else? If you want to learn anything about this game at all you have to read a wiki page or learn from someone else. Saying "thing in ss13 shouldnt be so complex in the first place" is like saying a hot dog is a sandwich.

Actually his point he explained later (unless he changed point mid-course) is that the wiki is more than enough and that new players will *obviously* go there on their own, which I disagree with. They mostly don't like the esthetic of the newbie tips popup.

**Comment by Eneocho** [comment]
Can't wait for Cult 5.

**Comment by donteviii** [comment]
I'm a hardcore jobbie that never plays antag, so I'm usually scrambling to the wiki page when I get converted to an antag role, especially cult. The effort to help newbie cult players is appreciated. I still don't know how cult 3.0 works.

On a different note, what sort of anti-cult features can we expect? Do janitors get any role in fighting cult 4.0?

**Comment by DeityLink** [comment]
> On a different note, what sort of anti-cult features can we expect? Do janitors get any role in fighting cult 4.0?

The anti-cult Cult 4 update was added back in May technically #29630, but now janitors can clean unfinished/non-activated runes with water alone. Job-focused activities will come with the Rituals update.

**Comment by cadaverouscoyote** [comment]
> And once again that's beside the point: which is that most players don't go on the wiki blind, and I argue they shouldn't.

i don't agree that the majority of players won't check the wiki when they need info on a role, considering there's yet another link to it before they start playing in the form of the "Role Wiki" buttons beneath the "Instructions" label next to each individual preference in the special roles setup window. it could not be made any clearer to the new player that they're supposed to read the wiki without needing a password from the page itself. i think you're fixating on your experiences with a proportionately small minority of the most thoughtless players and treating their own failures as failures of the game design. i wanna emphasize: somebody that stupid isn't gonna click the tutorial button without having to be told by someone, either

i also don't see any sense in discouraging the use of an external resource before a player's first experience. given that the very first link on the wiki's main page is the one that runs you through how to connect to the server in the first place, i don't think your vision is widely shared

> So you're purposefully misrepresentating your own arguments

don't get too hung up on semantics

> Well ok let's list a few then.

animations and instantaneousness aren't of much relevance to this thing's stated purpose and don't make up for the functionality that it's missing. if i needed to learn, i'd take a single .jpeg infographic that i could place anywhere that's convenient over all of that. the imgur link is currently a superior alternative

> How is that not what I'm doing?

just like i was concerned there'd be, there are a lot of tweaks included here that could have been separate pr's. i understand that you need to see how all of these things work in cohesion because that's how you would intend it to be down the line, but while you can drop as many changes at once as you want, you can only effectively further tweak a couple of them at once, max. merging, then adjusting one thing at a time is what will let players give you the most informed and to-the-point feedback possible. my position is that making multiple, simultaneous changes based on theorizing isn't the most reliable way of making constructive progress. i didn't say that nothing's worth changing, but that there's a needlessly arduous method of implementing changes, which can leave you doing cleanup longer than you ever had to, and i think that's evident

>I'll argue that the worst potential catastrophic fallout would still be limited to cult, and easy to contain even by just disabling the ruleset.

needing to outright disable a mode is what i'd consider a catastrophic failure, which would carry additional in-game repercussions

> They mostly don't like the esthetic of the newbie tips popup.

if it was a matter of taste alone, one post would've conveyed all of my thoughts. my issue with its execution is that it's pointlessly obstructive and awkward when the purpose is supposed to be accessibility, and my issue with it conceptually is that it accomplishes nothing that an existing thing can't

anyway, on the concept of freeform cult

i don't think it's going to ignite much long-lasting enthusiasm for the mode. if we let individual cultists exercise their antag freedoms however they please in the absence of an objective, then we're bound to have ablinos doing "rp" conversions to get handed their school shooter license, and thus sec treating even the most innocuous mom & pop candle shop as a murderboner factory. if, since cult is still a team mode, the rule is instead that insubordination is grounds for an antag ban, then some of the free has been taken out of freeform and i see many players still being unwilling to get pulled away from what they were doing just to get ordered around. i remain neutral

like i said though, only one way to know how anything will work in practice, so we'll see i guess

**Comment by Brotanics** [comment]
It's sad to see you go, Cult. Even if I wasn't the biggest fan.

**Comment by Eneocho** [comment]
Here you can observe what obsession does to a man. Sheer depravity. In 3 days he did the job of 12 days. May the void take pity on his soul.

**Comment by Eneocho** [comment]
Here's hoping cult 4 will be the last iteration of cult and that the cult 5 joke stays a joke.

**Comment by DeityLink** [comment]
> Here you can observe what obsession does to a man. Sheer depravity. In 3 days he did the job of 12 days. May the void take pity on his soul.

When you rebase a branch every past commits get brought along. So no, these 19 days are spread over the past 2 months.

> Here's hoping cult 4 will be the last iteration of cult and that the cult 5 joke stays a joke.

Why? Don't wanna joke about cult 6? If there's room for improvement and I'm still motivated in 3 years I see no reason not to. The numbers on those versions are arbitrary anyway, they just help us a bit know which versions of cult we're talking about in conversations. "Cult 1" was how I called cult as it was when I started playing but it definitely wasn't the first overall version of cult. Cult 2 added the objectives requirements to be able to raise Nar-Sie, Cult 3 added the ACT system and bloodstone phase, Cult 4's big gimmick will be the cult roles, the Ritual system, and the Eclipse. This PR only adds the first of those so it's definitely not the end of cult ""development"" by a long shot.

**Comment by Eneocho** [comment]
What I meant is that hopefully this iteration is good enough to stay for good. Improvements are great and all, but complete overhauls just show that the idea was poorly implemented in some way.

**Comment by DeityLink** [comment]
Well, this is it.

**Comment by DeityLink** [comment]
conflicts begone

File: code/_onclick/mindUI/bloodcult/draw_runes.dm [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+	var/image/word_overlay
+
+/obj/abstract/mind_ui_element/hoverable/rune_word/Appear()
+	..()
+	if (word && !mouse_opacity)
+		mouse_opacity = 1
+		flick("rune_appear",src)
+
+
+/obj/abstract/mind_ui_element/hoverable/rune_word/proc/get_word_order()
```
Comment (Kurfursten) [inline code/_onclick/mindUI/bloodcult/draw_runes.dm:18]: why do runes use this system instead of just storing words in a list? then you could at any time reference word[2] or Find(word)
